loader/loader.c

```diff
@@ -695,7 +698,11 @@ VkResult loaderGetDeviceRegistryFiles(const struct loader_instance *inst, char *
                                       LPCSTR value_name) {
     static const wchar_t *softwareComponentGUID = L"{5c4c3332-344d-483c-8739-259e934c9cc8}";
     static const wchar_t *displayGUID = L"{4d36e968-e325-11ce-bfc1-08002be10318}";
+#ifdef CM_GETIDLIST_FILTER_PRESENT
     const ULONG flags = CM_GETIDLIST_FILTER_CLASS | CM_GETIDLIST_FILTER_PRESENT;
+#else
+    const ULONG flags = CM_GET_DEVICE_INTERFACE_LIST_PRESENT;
```

Comment:
This is pretty clearly clear intended for the `CM_Get_Device_Interface_List` functions whereas we're using this in the `CM_Get_Device_ID_List` functions so I don't think this is quite correct. Interestingly, the value for this is zero, which is equivalent to `CM_GETIDLIST_FILTER_NONE` so it probably works ok. But that does mean you're not filtering out devices that have been removed (which is half the point we added this code). I think the solution I'd go with is probably just to add the numerical value instead so the behavior doesn't change. So I'd change this to:

```c
const ULONG flags = 0x300; // Equivalent to (CM_GETIDLIST_FILTER_CLASS | CM_GETIDLIST_FILTER_PRESENT) which aren't always defined on MinGW
```

Reply:
Yes, that makes more sense.
My attempt was to find the equivalent in MinGW-w64, but if it results in 0 that doesn't make sense.
Go ahead and change it if you like.